python/pyspark/ml/regression.py

```diff
@@ -1312,8 +1312,7 @@ def __init__(self, labelCol="label", featuresCol="features", predictionCol="pred
         super(GeneralizedLinearRegression, self).__init__()
         self._java_obj = self._new_java_obj(
             "org.apache.spark.ml.regression.GeneralizedLinearRegression", self.uid)
-        self._setDefault(family="gaussian", maxIter=25, tol=1e-6, regParam=0.0, solver="irls",
-                         linkPredictionCol="")
+        self._setDefault(family="gaussian", maxIter=25, tol=1e-6, regParam=0.0, solver="irls")
```

Comment:
I think the setDefault change you have is fine - I mean the default
constructor param
On Fri, 20 May 2016 at 11:37, Liang-Chi Hsieh <notifications@github.com>
wrote:

> In python/pyspark/ml/regression.py
> <https://github.com/apache/spark/pull/13220#discussion_r64014350>:
>
> > @@ -1312,8 +1312,7 @@ def __init__(self, labelCol="label", featuresCol="features", predictionCol="pred
> >          super(GeneralizedLinearRegression, self).__init__()
> >          self._java_obj = self._new_java_obj(
> >              "org.apache.spark.ml.regression.GeneralizedLinearRegression", self.uid)
> > -        self._setDefault(family="gaussian", maxIter=25, tol=1e-6, regParam=0.0, solver="irls",
> > -                         linkPredictionCol="")
> > +        self._setDefault(family="gaussian", maxIter=25, tol=1e-6, regParam=0.0, solver="irls")
>
> I was not sure about this. So just following other similar params in other
> classes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/13220/files/132753838a1424f09d85003a4436bb7ce0bd7774#r64014350>
>